value/src/main/java/com/google/auto/value/processor/AutoValueProcessor.java

```diff
@@ -112,6 +116,23 @@ public synchronized void init(ProcessingEnvironment processingEnv) {
     }
   }
 
+  @Override
+  public Set<String> getSupportedOptions() {
+    AutoValueExtension.IncrementalExtensionType incrementalType = extensions.stream()
+        .map(e -> e.incrementalType(processingEnv))
+        .min(Comparator.naturalOrder())
+        .orElse(AutoValueExtension.IncrementalExtensionType.ISOLATING);
+    switch (incrementalType) {
+      case ISOLATING:
+        return ImmutableSet.of(OMIT_IDENTIFIERS_OPTION, IncrementalAnnotationProcessorType.ISOLATING.getProcessorOption());
+      case AGGREGATING:
+        return ImmutableSet.of(OMIT_IDENTIFIERS_OPTION, IncrementalAnnotationProcessorType.AGGREGATING.getProcessorOption());
+      case UNKNOWN:
```

Comment:
I would delete this, or delete the default case and follow the switch with `throw new AssertionError(incrementalType)`.